File: BinarySearch/Q704.java
// https://leetcode.com/problems/binary-search/
public class Q704 {
    public int search(int[] nums, int target) {
        int lo = 0, hi = nums.length - 1;
        while (lo <= hi) {
            int mid = lo + (hi - lo) / 2;
            int cmp = Integer.compare(target, nums[mid]);
            if (cmp < 0)      hi = mid - 1;
            else if (cmp > 0) lo = mid + 1;
            else              return mid;
        }
        return -1;
    }
}

File: BinarySearch/Q704.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        lo, hi = 0, len(nums) - 1
        while lo <= hi:
            mid = (lo + hi) // 2
            cmp = target - nums[mid]
            if cmp < 0:
                hi = mid - 1
            elif cmp > 0:
                lo = mid + 1
            else:
                return mid
        return -1
